### ISSUE TITLE
Attachments are not updated (wrong date format)

### ISSUE THREAD

**Issue by Josef-Reichardt** [bsorrentino/maven-confluence-plugin#160]
After using this plugin for a while I realized that attachments are only created initially but never updated if they change.

I've found something like this in the logs:
```
[INFO] generateAttachment pageId [123456789] title [My Page] file [file:/img/logo.png]
[INFO] attachment [logo.png] skipped! no updated detected
```

The plugin always says `skipped! no updated detected`.

I've already done a little research:
The date which was loaded from the confluence REST API was parsed by this pattern: `"yyyy-MM-dd'T'HH:mm:ss'Z'"` ([`Attachment.java#L58`](https://github.com/bsorrentino/maven-confluence-plugin/blob/v5.1.1/maven-confluence-core/src/main/java/org/bsc/confluence/rest/model/Attachment.java#L58))
But the it was for example `"2018-07-03T13:35:11.450+02:00"` (BTW: the patter for this would be `"yyyy-MM-dd'T'HH:mm:ss.SSSX"` and is ISO 8601 valid). Because of the milliseconds and the timezone this leads to a `ParseException` ([`Attachment.java#L78-L80`](https://github.com/bsorrentino/maven-confluence-plugin/blob/v5.1.1/maven-confluence-core/src/main/java/org/bsc/confluence/rest/model/Attachment.java#L78-L80)) and a `new Date()` ([`Attachment.java#L81`](https://github.com/bsorrentino/maven-confluence-plugin/blob/v5.1.1/maven-confluence-core/src/main/java/org/bsc/confluence/rest/model/Attachment.java#L81)) is used as _"confluence last modified date"_.

This is my configuration in `pom.xml`:
```xml
<plugin>
    <groupId>org.bsc.maven</groupId>
    <artifactId>confluence-reporting-maven-plugin</artifactId>
    <version>${confluence-reporting-maven-plugin.version}</version>
    <configuration>
        <endPoint>https://my-confluence-host.net/rest/api</endPoint>
        <spaceKey>MYCONFLUENCESPACE</spaceKey>
        <parentPageId>12345678</parentPageId>
        <parentPageTitle>Parent</parentPageTitle>
        <serverId>confluence_server</serverId>
        <labels>
            <label>generated</label>
        </labels>
        <skip>${confluence-reporting-maven-plugin.skip}</skip>
    </configuration>
</plugin>
```

I've already tried to set the timezone in the confluence user settings to UTC but this seems to have no effect on the REST API (I tried it using `curl -D- -u confuser:confpass -X GET https://my-confluence-host.net/rest/api/content/123456789/child/attachment\?filename\=logo.png\&expand\=space,version,container`).

Is there anything I can do or is it a big in this plugin?

**Comment by bsorrentino** [comment]
Which version of plugin are you used?

In the latest version i don’t  use the date strategy anymore

**Comment by Josef-Reichardt** [comment]
I can't use v6.0-rc1 because it's not available in maven central.  
There is something missing: http://search.maven.org/#artifactdetails%7Corg.bsc.maven%7Cconfluence-reporting-maven-plugin%7C6.0-rc1%7Cmaven-plugin  
And there I got a 404: http://repo1.maven.org/maven2/org/bsc/maven/confluence-reporting-maven-plugin/6.0-rc1/

I've tried it with v6.0-beta2, but I got an error:
```
java.lang.Error: {"id":<...>}
	at org.bsc.confluence.rest.AbstractRESTConfluenceService.mapToArray(AbstractRESTConfluenceService.java:156)
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:267)
	...
```

Tomorrow I will look into it more detailed.

**Comment by bsorrentino** [comment]
@Josef-Reichardt thank you for your valuable feedbacks

Firstly I've just published the version `6.0-rc1`, please use it for all test.

ASAP, I'll set-up an environment equivalent yours ... to verify issue and eventually  provide  fix

**Comment by bsorrentino** [comment]
Using version `6.0-rc1` against confluence `5.1.0` the issue doesn't coming out

**Comment by Josef-Reichardt** [comment]
Sry, I'm not sure, but I think something is wrong here. Maybe I just don't see the magic but now I want to understand how this works.

1. The URL was something like this: `https://cwiki.apache.org/confluence/rest/api/content/260?expand=space,version,container` ([`AbstractRESTConfluenceService.java#L198-L202`](https://github.com/bsorrentino/maven-confluence-plugin/blob/develop/maven-confluence-core/src/main/java/org/bsc/confluence/rest/AbstractRESTConfluenceService.java#L198-L202))
2. And the result is always passed to the `mapToArray` function ([`AbstractRESTConfluenceService.java#L204`](https://github.com/bsorrentino/maven-confluence-plugin/blob/develop/maven-confluence-core/src/main/java/org/bsc/confluence/rest/AbstractRESTConfluenceService.java#L204))
3. And there is a check which throws an `Error` if there is no key `results` on root level ([`AbstractRESTConfluenceService.java#L155-L157`](https://github.com/bsorrentino/maven-confluence-plugin/blob/develop/maven-confluence-core/src/main/java/org/bsc/confluence/rest/AbstractRESTConfluenceService.java#L155-L157))

If I simply open the example URL from point 1 (https://cwiki.apache.org/confluence/rest/api/content/260?expand=space,version,container) in my Browser I get a JSON response but there is no key `results`.
Could you please explain it more clearly?
Or at least can you give me any hint what I can do to get the plugin (v6.0-rc1) working again?

**Comment by bsorrentino** [comment]
Hi @Josef-Reichardt 

I haven't used magic code ... 

Just (if I remember well) the **http client library** ( ie [okhttp](https://github.com/square/okhttp) ) **adds `results` array by default**

**Comment by Josef-Reichardt** [comment]
Sorry, I really don't want to tease you but I think that's not correct.
I've just copied all the required parts together in a simple demo project. See attached [example.zip](https://github.com/bsorrentino/maven-confluence-plugin/files/2283119/example.zip). If you run the `main` function you will get exactly the error as I when running the maven plugin.
Btw, if we find the reason for this issue I could provide a pull request if you want.

If you don't think that it's a bug, do you have any idea what I could try to get v6.0-rc1 working on my side?

**Comment by bsorrentino** [comment]
Hi @Josef-Reichardt 

Thank you for test code.  I'll try to figure out why on my side it works and on yours not

It MUST works !!!

**Comment by bsorrentino** [comment]
Hi @Josef-Reichardt

I've reproduced problem .... fix it and  deployed version `6.0-SNAPSHOT`

Can you test it and let me know ?

Thanks in advance

**Comment by Josef-Reichardt** [comment]
Great news, thank you very much. :slightly_smiling_face: 

I will try it tomorrow.

**Comment by bsorrentino** [comment]
Ok 

I'll wait your feedback for new `6.0-rc2` release

**Comment by bsorrentino** [comment]
fix released with `6.0-rc2` version
